--- a/CMake/third-party/FBThriftCppLibrary.cmake
+++ b/CMake/third-party/FBThriftCppLibrary.cmake
@@ -44,7 +44,7 @@
   # Generate relative paths in #includes
   file(
     RELATIVE_PATH include_prefix
-    "${CMAKE_SOURCE_DIR}"
+    "${PROJECT_SOURCE_DIR}"
     "${CMAKE_CURRENT_SOURCE_DIR}/${THRIFT_FILE}"
   )
   get_filename_component(include_prefix ${include_prefix} DIRECTORY)
@@ -123,7 +123,7 @@
       -o "${output_dir}"
       "${CMAKE_CURRENT_SOURCE_DIR}/${THRIFT_FILE}"
     WORKING_DIRECTORY
-      "${CMAKE_BINARY_DIR}"
+      "${PROJECT_BINARY_DIR}"
     MAIN_DEPENDENCY
       "${THRIFT_FILE}"
     DEPENDS
@@ -136,7 +136,7 @@
   target_include_directories(
     "${LIB_NAME}"
     PUBLIC
-      "$<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>"
+      "$<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>"
       "$<INSTALL_INTERFACE:${ARG_INCLUDE_DIR}>"
       ${Xxhash_INCLUDE_DIR}
   )
@@ -173,7 +173,7 @@
   target_include_directories(
     "${LIB_NAME}.thrift_includes"
     INTERFACE
-      "$<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>"
+      "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>"
       "$<INSTALL_INTERFACE:${ARG_THRIFT_INCLUDE_DIR}>"
   )
   foreach(dep IN LISTS ARG_DEPENDS)
