Index: velox/vector/BaseVector.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/velox/vector/BaseVector.cpp b/velox/vector/BaseVector.cpp
--- a/velox/vector/BaseVector.cpp	(revision 9f7f64abdfc6f078adfbc0d1b07de0399f38d482)
+++ b/velox/vector/BaseVector.cpp	(revision 91c39aac46f040b574986a8240c21d277f12b72c)
@@ -832,7 +832,14 @@
     using T = typename TypeTraits<KIND>::NativeType;

     const vector_size_t dataSize = data.size();
-    BufferPtr valuesBuffer = AlignedBuffer::allocate<T>(dataSize, pool);
+    BufferPtr valuesBuffer;
+    if constexpr (std::is_same_v<T, StringView>) {
+      // Make sure to initialize StringView values so they can be safely
+      // accessed.
+      valuesBuffer = AlignedBuffer::allocate<T>(dataSize, pool, T());
+    } else {
+      valuesBuffer = AlignedBuffer::allocate<T>(dataSize, pool);
+    }
     BufferPtr nulls = allocateNulls(dataSize, pool, bits::kNull);

     auto values = std::make_shared<FlatVector<T>>(
